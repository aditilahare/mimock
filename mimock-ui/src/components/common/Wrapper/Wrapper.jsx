import React from 'react';
import PropTypes from 'prop-types';
import Header from 'components/common/Header';

function Wrapper({ children }) {
	return (
		<>
			<Header />
			{children}
		</>
	);
}

Wrapper.propTypes = {
	children: PropTypes.node.isRequired,
};

export default Wrapper;
