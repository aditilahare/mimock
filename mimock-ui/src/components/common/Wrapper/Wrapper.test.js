import React from 'react';
import { render } from '@testing-library/react';
import Wrapper from './Wrapper';

jest.mock('react-router-dom');

describe('Wrapper', () => {
	it('should render wrapper', async () => {
		const { getByTestId, container } = await render(
			<Wrapper>
				<div data-testid='wrapped-component'>Wrapped child</div>
			</Wrapper>
		);

		expect(getByTestId('wrapped-component')).toBeInTheDocument();
		expect(getByTestId('header')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
