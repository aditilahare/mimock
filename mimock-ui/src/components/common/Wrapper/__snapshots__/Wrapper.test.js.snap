// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render wrapper 1`] = `
<div>
  <div
    class="w-full h-20 bg-white flex items-center px-4 py-2 shadow-md justify-space-evenly mx-auto"
    data-testid="header"
  >
    <div
      class="flex flex-row items-center justify-between w-full"
    >
      <div
        class="mx-10 flex items-center text-gray-600 gap-10"
        data-testid="header-menu-items"
      />
      <div
        class="flex flex-row gap-8 items-center justify-center text-2xl text-gray-500"
        data-testid="header-menu-options"
      />
    </div>
  </div>
  <div
    data-testid="wrapped-component"
  >
    Wrapped child
  </div>
</div>
`;
